src/battle_dome.c

```diff
@@ -4130,6 +3928,244 @@ static u8 Task_GetInfoCardInput(u8 taskId)
+    default:
+        return FALSE;
+    }
+}
+
+static bool32 IsDomeDefensiveMoveEffect(u32 effect)
+{
+    switch(effect)
+    {
+    case EFFECT_COUNTER:
+    case EFFECT_EVASION_UP:
+    case EFFECT_DEFENSE_UP:
+    case EFFECT_DEFENSE_UP_2:
+    case EFFECT_SPECIAL_DEFENSE_UP:
+    case EFFECT_SPECIAL_DEFENSE_UP_2:
+    case EFFECT_MINIMIZE:
+    case EFFECT_ACCURACY_DOWN:
+    case EFFECT_DEFENSE_CURL:
+    case EFFECT_LIGHT_SCREEN:
+    case EFFECT_REFLECT:
+    case EFFECT_AURORA_VEIL:
+    case EFFECT_CONVERSION:
+    case EFFECT_PROTECT:
+    case EFFECT_MAT_BLOCK:
+    case EFFECT_ENDURE:
+    case EFFECT_SAFEGUARD:
+    case EFFECT_MIRROR_COAT:
+    case EFFECT_MAGIC_COAT:
+    case EFFECT_INGRAIN:
+    case EFFECT_AQUA_RING:
+    case EFFECT_SUBSTITUTE:
+        return TRUE;
+    default:
+        return FALSE;
+    }
+}
+
+static bool32 IsDomeRiskyMoveEffect(u32 effect)
+{
+    switch(effect)
+    {
+    case EFFECT_EXPLOSION:
+    case EFFECT_SPITE:
+    case EFFECT_DESTINY_BOND:
+    case EFFECT_PERISH_SONG:
+        return TRUE;
+    default:
+        return FALSE;
+    }
+}
+
+static bool32 IsDomeLuckyMove(u32 move)
+{
+    if (gBattleMoves[move].accuracy <= 50)
+        return TRUE;
+    switch(gBattleMoves[move].effect)
+    {
+    case EFFECT_COUNTER:
+    case EFFECT_OHKO: // Technically redundant because of the above accuracy check
+    case EFFECT_METRONOME:
+    case EFFECT_MIRROR_MOVE:
+    case EFFECT_SKETCH:
+    case EFFECT_SLEEP_TALK:
+    case EFFECT_PRESENT:
+    case EFFECT_ASSIST:
+    case EFFECT_MAGIC_COAT:
+    case EFFECT_REVENGE:
+    case EFFECT_IMPRISON:
+    case EFFECT_SNATCH:
+        return TRUE;
+    default:
+        return FALSE;
+    }
+}
+
+static bool32 IsDomePopularMove(u32 move)
+{
+    u8 i;
+    for (i = 0; i < NUM_TECHNICAL_MACHINES + NUM_HIDDEN_MACHINES; i++)
+    {
+        if (ItemIdToBattleMoveId(ITEM_TM01 + i) == move)
+            return TRUE;
+    }
+    if (i == NUM_TECHNICAL_MACHINES + NUM_HIDDEN_MACHINES)
+        return FALSE;
+    // Filter in TMs/HMs
+    if (gBattleMoves[move].power >= 90)
+        return TRUE;
+
+    switch(gBattleMoves[move].effect)
+    {
+    case EFFECT_PROTECT:
+    case EFFECT_MAT_BLOCK:
+    case EFFECT_ATTACK_UP_2:
+    case EFFECT_SPECIAL_ATTACK_UP_2:
+    case EFFECT_SPECIAL_ATTACK_UP_3:
+        return TRUE;
+    default:
+        return FALSE;
+    }
+}
+
+static bool32 IsDomeStatusMoveEffect(u32 effect)
+{
+    switch(effect)
+    {
+    case EFFECT_TRAP:
+    case EFFECT_SLEEP:
+    case EFFECT_CONFUSE:
+    case EFFECT_DISABLE:
+    case EFFECT_POISON:
+    case EFFECT_PARALYZE:
+    case EFFECT_TOXIC:
+    case EFFECT_LEECH_SEED:
+    case EFFECT_TAUNT:
+    case EFFECT_TORMENT:
+    case EFFECT_WILL_O_WISP:
+    case EFFECT_ENCORE:
+    case EFFECT_ATTRACT:
+    case EFFECT_NIGHTMARE:
+    case EFFECT_YAWN:
+    case EFFECT_CURSE:
+        return TRUE;
+    default:
+        return FALSE;
+    }
+}
+
+static bool32 IsDomeRareMove(u32 move)
```

Comment:
It's pretty damn arbitrary in vanilla, so I had to make a choice. Future Sight seems to have been the most common level-up move with 20 occurrences (out of 386), but since my math skills seems to have vanished 5% would be much more accurate than 10% indeed. I'll make the change!

Reply:
Makes sense, though I wonder if we should check TM availability as well 🤔

Reply:
Hidden Power is considered rare, even if it's a universal TM, so that's why I opted to only check level-up moves